src/filter/mod.rs

```diff
@@ -236,7 +251,14 @@ pub fn apply_to_commit(
             return Ok(id);
         }
 
-        for (f, i) in transaction.get_missing() {
+        let missing = transaction.get_missing();
+
+        let minimal_nesting = missing.get(0).map(|(x, _, _)| *x).unwrap_or(0);
```

Comment:
maybe a comment saying that this assumes missing are sorted by nesting (which is true)